src/resolve/resolved-dns-transaction.c

```diff
@@ -840,10 +840,8 @@ static void dns_transaction_cache_answer(DnsTransaction *t) {
                       dns_transaction_key(t),
                       t->answer_rcode,
                       t->answer,
-                      DNS_PACKET_CD(t->received) ? t->received : NULL, /* only cache full packets with CD on,
-                                                                        * since our use case for caching them
-                                                                        * is "bypass" mode which is only
-                                                                        * enabled for CD packets. */
+                      /* If neither DO nor EDE is set, the full packet isn't useful to cache */
+                      DNS_PACKET_DO(t->received) || t->answer_ede_rcode > 0 || t->answer_ede_msg ? t->received : NULL,
```

Comment:
hmm, hmm what does this have to do with EDE? can you explain?

Reply:
There were some tests failing before I added this test usage of the cached packet to forward EDE errors. Previously with dnssec enabled most requests from the stub have +cd so most packets were cached and this test did not fail, but with the updated condition it did.